decentraland/screens/BuyLandScreen.js

```diff
@@ -110,6 +119,11 @@ export class BuyLandScreen extends React.Component {
         gasParams
       );
 
+      const actionId = await action.getId();
+
+      // Assumes that the asset was already added to the MyLand list (Optimistic UI update)
```

Comment:
If we wanted to make the order independent, we could just return the `action` from this function so that we could use it outside of the scope of this function in a spot where the MyLandList optimistic update had already clearly happened based on the high-level control flow of functions that are called.

Reply:
Done.

Reply:
Thanks. Resolving this.